solvers/optimization_utils.h

```diff
@@ -0,0 +1,60 @@
+#include "drake/solvers/mathematical_program.h"
+#include "drake/solvers/decision_variable.h"
+#include "drake/math/autodiff.h"
+#include "drake/math/autodiff_gradient.h"
+
+namespace dairlib {
+namespace solvers {
+
+/// Check if the generic (nonlinear) constraints are satisfied up to the
+/// specified tolerance.
+/// Prints out any violating constraints
+/// @return True if all generic constraints are satisfied
+bool CheckConstraints(const drake::solvers::MathematicalProgram& prog,
```

Comment:
I was merging my branch with this branch, and I found that this seems to be a bug. Could either change this to `void CheckGenericConstraints` or modify the cc file so that the function returns a boolean variable.

Reply:
Good catch. Changed to `bool CheckGenericConstraints`, does that work for you?

Reply:
I only spotted this bug and actually do not have code to test it. By reading the updated code, I think it's correct!